pkg/provider/provider.go

```diff
@@ -350,22 +382,54 @@ func (p *Provider) getKeyVaultObjectContent(ctx context.Context, kvClient *kv.Ba
 
 	switch kvObject.ObjectType {
 	case VaultObjectTypeSecret:
-		return getSecret(ctx, kvClient, vaultURL, kvObject)
+		return getSecret(ctx, kvClient, vaultURL, kvObject, objectVersion)
 	case VaultObjectTypeKey:
-		return getKey(ctx, kvClient, vaultURL, kvObject)
+		return getKey(ctx, kvClient, vaultURL, kvObject, objectVersion)
 	case VaultObjectTypeCertificate:
-		return getCertificate(ctx, kvClient, vaultURL, kvObject)
+		return getCertificate(ctx, kvClient, vaultURL, kvObject, objectVersion)
 	default:
 		err := errors.Errorf("Invalid vaultObjectTypes. Should be secret, key, or cert")
-		return "", "", wrapObjectTypeError(err, kvObject.ObjectType, kvObject.ObjectName, kvObject.ObjectVersion)
+		return "", "", wrapObjectTypeError(err, kvObject.ObjectType, kvObject.ObjectName, objectVersion)
+	}
+}
+
+func (p *Provider) getKeyVaultSecretVersions(ctx context.Context, kvClient *kv.BaseClient, kvObject KeyVaultObject, vaultURL string) (secretVersions []string, err error) {
+	start := time.Now()
+	defer func() {
+		var errMsg string
+		if err != nil {
+			errMsg = err.Error()
+		}
+		p.reporter.ReportKeyvaultRequest(ctx, time.Since(start).Seconds(), kvObject.ObjectType, kvObject.ObjectName, errMsg)
+	}()
+
+	versions, err := kvClient.GetSecretVersions(ctx, vaultURL, kvObject.ObjectName, &kvObject.ObjectVersionHistory)
```

Comment:
I'm not sure if this will actually give you back the top `N`, will it?

Top based on which criteria, I would say the `creation-time`, but couldn't find documentation explaining the ordering criteria, if any, does the list API support.

Reply:
that's a good point. my intention is top by `creation-time`, and my assumption is that the API was returning things in that order - i'll look into the API and figure out how to make that work appropriately

I'm currently working on getting this change deployed to one of my teams clusters, so I'll be able to test out a few different scenarios I have in mind

Reply:
i asked my team and they said it wasn't safe to trust the ordering returned from the kv api, and so I added some logic to explicitly order based on the creation date